### PR TITLE
DT-2632 Generic service chart updates

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: helm_deploy/hmpps-manage-users/values.yaml [hunk 1]
```diff
@@ -1,5 +1,42 @@
...
+      nginx.ingress.kubernetes.io/limit-rpm: "120"
+      # setting to 429 (default is 503) so we can alert on this
+      nginx.ingress.kubernetes.io/server-snippet: |
+        add_header X-Robots-Tag "noindex, nofollow";
+        limit_req_status 429;
+    tlsSecretName: hmpps-manage-users-cert
+    modsecurity_enabled: true
+
+  env:
+    JAVA_OPTS: "-Xmx512m"
```
Comment (petergphillips) [inline helm_deploy/hmpps-manage-users/values.yaml:25]: these are the java versions, not suitable for node

File: helm_deploy/hmpps-manage-users/values.yaml [hunk 2]
```diff
@@ -1,5 +1,42 @@
...
+  # namespace_secrets:
+  #   [name of kubernetes secret]:
+  #     [name of environment variable as seen by app]: [key of kubernetes secret to load]
+  namespace_secrets:
+    hmpps-manage-users:
+      API_CLIENT_ID: "API_CLIENT_ID"
+      API_CLIENT_SECRET: "API_CLIENT_SECRET"
+      GOOGLE_TAG_MANAGER_ID: "GOOGLE_TAG_MANAGER_ID"
+      SESSION_COOKIE_SECRET: "SESSION_COOKIE_SECRET"
+      REDIS_HOST: "REDIS_HOST"
```
Comment (petergphillips) [inline helm_deploy/hmpps-manage-users/values.yaml:38]: this should be from the `hmpps-redis` secret instead

File: helm_deploy/values-dev.yaml [hunk 3]
```diff
@@ -1,38 +1,27 @@
...
-# Declare variables to be passed into your templates.
+---
+# Environment specific values, override helm_deploy/hmpps-manage-users/values.yaml
 
-replicaCount: 2
+generic-service:
+  replicaCount: 2
+  ingress:
+    host: manage-users-dev.hmpps.service.justice.gov.uk
+    modsecurity_audit_enabled: true
```
Comment (petergphillips) [inline helm_deploy/values-dev.yaml:8]: would be better to put this in `values.yaml` instead

File: helm_deploy/values-dev.yaml [hunk 4]
```diff
@@ -1,38 +1,27 @@
...
-  tag: latest
-  pullPolicy: IfNotPresent
-  port: 3000
+  env:
+    API_ENDPOINT_URL: https://api-dev.prison.service.justice.gov.uk
+    MANAGE_USERS_API_ENDPOINT_URL: https://hmpps-manage-users-api-dev.hmpps.service.justice.gov.uk
+    OAUTH_ENDPOINT_URL: https://sign-in-dev.hmpps.service.justice.gov.uk/auth
+    DPS_ENDPOINT_URL: https://digital-dev.prison.service.justice.gov.uk
+    HMPPS_COOKIE_NAME: hmpps-session-dev
+    NODE_ENV: production
```
Comment (petergphillips) [inline helm_deploy/values-dev.yaml:16]: some of these look identical across envs, so would be better in `values.yaml` instead

File: helm_deploy/values-preprod.yaml [hunk 5]
```diff
@@ -1,37 +1,26 @@
-# This is a YAML-formatted file.
-# Declare variables to be passed into your templates.
+---
+# Environment specific values, override helm_deploy/hmpps-manage-users/values.yaml
 
-replicaCount: 4
+generic-service:
+  replicaCount: 4
```
Comment (petergphillips) [inline helm_deploy/values-preprod.yaml:5]: default for replicas is 4 in the generic service I think

Comment (karenmillermoj) [reply helm_deploy/values-preprod.yaml:5]: Looks like it's 2 in values.yaml (maybe we should change it to 4?)

File: helm_deploy/values-prod.yaml [hunk 6]
```diff
@@ -1,40 +1,20 @@
...
-  tag: latest
-  pullPolicy: IfNotPresent
-  port: 3000
+  env:
+    API_ENDPOINT_URL: https://api.prison.service.justice.gov.uk
+    MANAGE_USERS_API_ENDPOINT_URL: https://hmpps-manage-users-api.hmpps.service.justice.gov.uk
+    OAUTH_ENDPOINT_URL: https://sign-in.hmpps.service.justice.gov.uk/auth
+    DPS_ENDPOINT_URL: https://digital.prison.service.justice.gov.uk
+    HMPPS_COOKIE_NAME: hmpps-session-prod
+    WEB_SESSION_TIMEOUT_IN_MINUTES: "60"
```
Comment (petergphillips) [inline helm_deploy/values-prod.yaml:14]: this could have gone in the values.yaml too

**Review by petergphillips — DISMISSED** [review]
LGTM

File: helm_deploy/values-prod.yaml (hunk 6 above)
Comment (karenmillermoj) [reply helm_deploy/values-prod.yaml:14]: you are right - will make a note and push back with next PR

**Pull request merged**